### PR TITLE
encoderd: enable

### PR DESCRIPTION
**Pull request by geohot**
All functionality is in master, just have to make the tests pass for this. (I think it's good)

encoderd is usually the process driving the rotation unless loggerd has not received any encoder packets in the last 500ms. The log will rotate once all 4 encoders have.

1200 frame based rotation logic remains the same, it's just now in encoderd. The rotate_cv has been replaced with a queue which enqueues the encoder messages for the next segment.

### CODE REVIEW

File: selfdrive/hardware/tici/test_power_draw.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ class Proc:
   Proc('camerad', 2.5),
   Proc('modeld', 0.95),
   Proc('dmonitoringmodeld', 0.25),
-  Proc('loggerd', 0.45, warmup=10.),
+  Proc('encoderd', 0.42),
```
Comment (geohot) [inline selfdrive/hardware/tici/test_power_draw.py:25]: Half of this power is the NV12 copy!

**Comment by geohot** [comment]
Test failure is real, will investigate tomorrow.

File: selfdrive/loggerd/encoderd.cc [hunk 2]
```diff
@@ -139,8 +140,8 @@ int main() {
     int ret;
     ret = util::set_realtime_priority(52);
     assert(ret == 0);
-    ret = util::set_core_affinity({7});
-    assert(ret == 0);
+    ret = util::set_core_affinity({3});
+    assert(ret == 0 || Params().getBool("IsOffroad"));
```
Comment (adeebshihadeh) [inline selfdrive/loggerd/encoderd.cc:144]: this can be removed since core 3 is offline while offroad

**Comment by geohot** [comment]
Woot, jenkins passes!

**Comment by geohot** [comment]
So this is now hitting the qcam size bug.

@adeebshihadeh can you look into this since it's happening on master also? I suspect it's what the fcamera is seeing, and the new devices see different things

**Comment by geohot** [comment]
Be aware, this changes behavior slightly since encoderd is the primary driver of log rotation now. It shouldn't matter in the car since loggerd is started before either encoderd or camerad (just one or the other, with camerad latency this shouldn't ever happen)

But if it does, the first segment won't contain 1200 frames. The first segment also won't contain 1200 frames when you start and kill the logger on the body.

**Pull request merged**